controllers/issuer_controller.go

```diff
@@ -69,51 +76,58 @@ type IssuerReconciler struct {
 	PostSetupWithManager func(context.Context, schema.GroupVersionKind, ctrl.Manager, controller.Controller) error
 }
 
-func (r *IssuerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, returnedError error) {
+func (r *IssuerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	logger := log.FromContext(ctx).WithName("Reconcile")
 
 	logger.V(2).Info("Starting reconcile loop", "name", req.Name, "namespace", req.Namespace)
 
-	result, issuerStatusPatch, returnedError := r.reconcileStatusPatch(logger, ctx, req)
-	logger.V(2).Info("Got StatusPatch result", "result", result, "patch", issuerStatusPatch, "error", returnedError)
-	if issuerStatusPatch != nil {
-		cr, patch, err := ssaclient.GenerateIssuerStatusPatch(r.ForObject, req.Name, req.Namespace, issuerStatusPatch)
-		if err != nil {
-			return ctrl.Result{}, utilerrors.NewAggregate([]error{err, returnedError})
-		}
+	result, issuerStatusPatch, reconcileError := r.reconcileStatusPatch(logger, ctx, req)
+
+	if issuerStatusPatch == nil {
+		logger.V(2).Info("Got nil StatusPatch result", "result", result, "error", reconcileError)
+		return ctrl.Result{}, reconcileError
+	}
```

Comment:
**Question:** This block is confusing me as I do not understand how it is possible to have simultaneously `issuerStatusPatch == nil` and `reconcileError == nil`. I have left a comment above on the func `reconcileStatusPatch` to learn more.

Reply:
Good call. I tried to flatten this if case, but that made the logic a lot more confusing. Hopefully it is clear now (I also added a comment here: https://github.com/cert-manager/issuer-lib/pull/42/files#diff-4baeeaeddf38ed0df2790cd1e77c402812e8cc07f759ca675a83d271b013078dR110-R113